portality/bll/doaj.py

```diff
@@ -0,0 +1,620 @@
+            {"arg" : account, "instance" : models.Account, "allow_none" : False, "arg_name" : "account"}
+        ], exceptions.ArgumentException)
+
+        # get hold of a copy of the application.  If there isn't one, our work here is done
+        # (note the application could be locked, in which case this will raise a lock.Locked exception)
+
+        # get the application
+        application, _ = self.application(application_id)
+        if application is None:
+            raise exceptions.NoSuchObjectException
+
+        # determine if the user can edit the application
+        self.can_edit_application(account, application)
+
+        # attempt to lock the record (this may raise a Locked exception)
+        alock = lock.lock(constants.LOCK_APPLICATION, application_id, account.id)
+
+        # obtain the current journal, with associated lock
+        current_journal = None
+        cjlock = None
+        if application.current_journal is not None:
+            try:
+                current_journal, cjlock = self.journal(application.current_journal, lock_journal=True, lock_account=account)
+            except lock.Locked as e:
+                # if the resource is locked, we have to back out
+                if alock is not None: alock.delete()
+                raise
+
+        # obtain the related journal, with associated lock
+        related_journal = None
+        rjlock = None
+        if application.related_journal is not None:
+            try:
+                related_journal, rjlock = self.journal(application.related_journal, lock_journal=True, lock_account=account)
+            except lock.Locked as e:
+                # if the resource is locked, we have to back out
+                if alock is not None: alock.delete()
+                if cjlock is not None: cjlock.delete()
+                raise
+
+        try:
+            if current_journal is not None:
+                current_journal.remove_current_application()
+                saved = current_journal.save()
+                if saved is None:
+                    raise exceptions.SaveException("Unable to save journal record")
+
+            if related_journal is not None:
+                relation_record = related_journal.related_application_record(application_id)
+                if relation_record is None:
+                    relation_record = {}
+                related_journal.add_related_application(application_id, relation_record.get("date_accepted"), "deleted")
+                saved = related_journal.save()
+                if saved is None:
+                    raise exceptions.SaveException("Unable to save journal record")
+
+            application.delete()
+
+        finally:
+            if alock is not None: alock.delete()
+            if cjlock is not None: cjlock.delete()
+            if rjlock is not None: rjlock.delete()
+
+        return
+
+    def journal(self, journal_id, lock_journal=False, lock_account=None, lock_timeout=None):
```

Comment:
I pondered that for a while when I created this function, and decided against get_journal since this form of function is typical of an accessor on a model object, and this is a functional aspect of the business layer.  Since this function does more than getting the journal (it can check and create locks, for instance), I decided to simply call the function "journal".

Reply:
To compare, other systems I've worked on used naming conventions like "findX" in this circumstance, rather than "getX", but I didn't particularly like that convention either, though it also represents a similar thought regarding separating the naming style from regular attribute accessors.